gcpvault.go

```diff
@@ -433,46 +425,71 @@ func newJWTBase(ctx context.Context, cfg Config) (string, error) {
 			Base:   hc.Transport,
 		},
 	}
-	iamClient, err := iam.New(hcIAM)
+
+	claim, err := json.Marshal(map[string]interface{}{
+		"aud": "vault/" + cfg.Role,
+		"sub": serviceAccount,
+		"exp": time.Now().UTC().Add(5 * time.Minute).Unix(),
+	})
+
 	if err != nil {
-		return "", errors.Wrap(err, "unable to init IAM client")
+		return "", errors.Wrap(err, "unable to encode JWT payload")
 	}
 
+	gcpURL := "https://iamcredentials.googleapis.com/v1"
 	if cfg.IAMAddress != "" {
-		iamClient.BasePath = cfg.IAMAddress
+		gcpURL = cfg.IAMAddress
 	}
+	// this additional marshalling is to escape the json
+	payload, err := json.Marshal(string(claim))
+	if err != nil {
+		return "", errors.Wrap(err, "unable to encode JWT payload")
+	}
+	body := []byte("{\"payload\":" + string(payload) + "}")
+	url := fmt.Sprintf(gcpURL+"/projects/-/serviceAccounts/%s:signJwt", serviceAccount)
 
-	resp, err := iamClient.Projects.ServiceAccounts.SignJwt(
-		fmt.Sprintf("projects/%s/serviceAccounts/%s",
-			project, serviceAccount),
-		&iam.SignJwtRequest{Payload: string(payload)}).Context(ctx).Do()
+	resp, err := hcIAM.Post(url, "application/json", bytes.NewBuffer(body))
+	if err != nil {
+		return "", errors.Wrap(err, "unable to POST")
+	}
+
+	defer resp.Body.Close()
+
+	body, readErr := ioutil.ReadAll(resp.Body)
+	if readErr != nil {
+		return "", errors.Wrap(err, "unable to parse response")
```

Comment:
here we are evaluating `readErr` but returning `err`...

can we use the `err` variable instead of creating a new error variable?